### ISSUE TITLE
Allocate/Initialization Naming

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/picongpu#1333]
Name all allocation and initialization members correctly.

- `allocate()`:
   - allocate ressources (such as GPUs) and memory
   - register in `DataConnector`
- `initialize()`:
   - fill allocated memory with data
   - add exchanges
- `clear()`:
   - reset to state after `allocate()` so `initialize()` can be called again (does not deallocate memory but might for example drop frames with particles)
- `deallocate()`:
   - deallocate memory
   - unsign from `DataConnector` (not implemented)

For functions that are currently not splitted or not useful to be splitted between `allocate()` and `initialize()` call them
- `allocateAndInitialize()`

**Comment by ax3l** [comment]
ccing @psychocoderHPC

**Comment by erikzenker** [comment]
I would like to have an offgithub meeting about this issue.

**Comment by ax3l** [comment]
Absolutely, just recorded our offline discussion results so far for continuation (-> [WIP])

**Comment by bussmann** [comment]
Would like to join.
